hl7/4.2/modules/ROOT/edits/hl7-connector-config-topics.adoc

```diff
@@ -0,0 +1,269 @@
+segment from the `basedefs.esl` file. You can then add the segments key
+following your message structure definition, followed by one or more
+segment definitions.
+
+=== Set Your Schema Locations
+
+You can configure schema locations in the Anypoint Studio XML view.
+
+. Switch to the XML view by clicking *Configuration XML*.
+
+. Modify your HL7 EDI configuration to include a list of all the schemas to include by adding an `+<http://edischema[edi:schema]>+` element for each document type:
++
+[source,xml,linenums]
+----
+<hl7-edi:config name="HL7_EDI__Configuration" identKeys="true" doc:name="HL7 EDI: Configuration">
+  <hl7-edi:schemas>
+    <hl7-edi:schema>hl7/v2_6/ADT_A01.esl</hl7-edi:schema>
+  </hl7-edi:schemas>
+</hl7-edi:config>
+----
++
+. After you create a global element for your HL7 EDI, configure the message structure, operations, and acknowledgments.
+
+== Event and Message to Message Structure Map
+
+If you configure the connector with multiple message structure schemas
+(whether in separate schema definitions files, as with the provided schemas,
+or in a single file) you need to define a mapping from the HL7 event
+and message types to message structures.
+
+HL7 defines the Message Type in the component values of `MSH-09`, a composite
+structure of type MSG. HL7 EDI Connector uses these component values to find the structure schema to use for processing a receive message, according to the following rules:
+
+* If the MSG-01 Message Code value is `ACK`, always use the predefined ACK schema.
+* If the MSG-03 Message Structure value is present (a value such
+as `ADT_A01`), use the schema structure with that ID.
+* If neither of these, then use a configured Event and Message to Message Structure Map to determine
+the message structure from the specified Trigger Event (`MSG-02`) and Message Code (`MSG-01`) values.
+
+The Event and Message to Message Structure Map is an optional configuration parameter.
+It must be a YAML file consisting of a map from each Event Type to a map for each
+supported Message Code to the actual Message Structure.
+
+Here's an example:
+
+[source,yaml,linenums]
+----
+A01: { XYZ: ADT_A01, ACK: ACK }
+A02: { XYZ: ADT_A02, ACK: ACK }
+A03: { XYZ: ADT_A03, ACK: ACK }
+A04: { XYZ: ADT_A01, ACK: ACK }
+A05: { XYZ: ADT_A05, ACK: ACK }
+A06: { XYZ: ADT_A06, ACK: ACK }
+A07: { XYZ: ADT_A06, ACK: ACK }
+A08: { XYZ: ADT_A01, ACK: ACK }
+----
+
+Each version of HL7 defines a different set of mappings from the event type and
+message code to the message structure. The default mappings are provided in the
+same JAR as the standard HL7 schema definitions, in files named `event-message.yaml`.
+You use the same type of paths for these mapping definitions as for the actual message structure schemas.
+
+=== General Tab
+
+Use the *General* tab to configure settings for reading and writing HL7 messages:
+
+* HL7 character encoding, always used for writing messages and used
```

Comment:
GLOBALLY evaluate list items for scanability. Break after the first element in an item, if it has a subject and then a definition or explanation, per SR:
```suggestion
* HL7 character encoding
+
Always used for writing messages. Used
```